packages/design-system/src/components/N8nButton/Button.vue

```diff
@@ -1,9 +1,11 @@
 <template>
-	<button
+	<component
+		:is="element"
```

Comment:
I think this would be clearer if the `<a>` "button" was a separate component for couple of reasons. 1) when I use a component name `Button` I'd imagine it's a `<button>` 2) now we have props like `href` which doesn't make sense for a `<button>`.

So maybe we could have a `N8nLinkButton` or so that looks like a button, but is an `<a>` element with `<a>` props. These could have the same base class if DRY is a concern

Reply:
Yeah, DRY was my main concern here since it looks like the `Button` component still has a lot for styling in the component itself and didn't want to add more scope to this by refactoring it. So I decided to take this route (taken from other frameworks like Quasar or Element UI).
Agree about the `href` validation will add that and will change the `link` prop to be string with values `button`, `a` but I think `elemnent` is more clear name for it then `as`